go-auto-instrumented/frontend/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.20 AS build
+WORKDIR /src
+ENV CGO_ENABLED=0
+
+COPY go.mod .
+COPY frontend/. .
+RUN go get
+RUN go build -o /out/frontend .
+
+FROM scratch AS bin
+WORKDIR /app
+COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+COPY --from=build /out/frontend /app/
+ENV MESSAGE_ENDPOINT=http://message-go-svc:9000
+ENV NAME_ENDPOINT=http://name-go-svc:8000
+ENV YEAR_ENDPOINT=http://year-go-svc:6001
```

Comment:
Do we need to set these as they are in the docker compose file too?

Reply:
this has been cleaned up now; only the necessary endpoints are included in Dockerfiles where appropriate, and can also be overwritten in k8s during runtime. some comments are also added to indicate why they are there - for example, if running the app locally with no env var set, it will use localhost. if running the docker image, it will use name:8000 unless overwritten at runtime with k8s or docker-compose. only having it in docker-compose means it's available at runtime if run with docker-compose, but not available at build time in the docker image itself.